cmd/utils/cmd.go

```diff
@@ -287,6 +290,54 @@ func ExportAppendChain(blockchain *core.BlockChain, fn string, first uint64, las
 	return nil
 }
 
+func ExportHistory(bc *core.BlockChain, dir string, first, last, step uint64) error {
+	log.Info("Exporting blockchain history", "dir", dir)
+	if head := bc.CurrentBlock().NumberU64(); head < last {
+		log.Warn("Last block beyond head, setting last = head", "head", head, "last", last)
+		last = head
+	}
+	network := "unknown"
+	if name, ok := params.NetworkNames[bc.Config().ChainID.String()]; ok {
+		network = name
+	}
+	for i := uint64(0); i < last; i += step {
+		// Open file for Era.
+		fn := path.Join(dir, fmt.Sprintf("%s-%05d.era", network, i/step))
+		fh, err := os.OpenFile(fn, os.O_CREATE|os.O_APPEND|os.O_WRONLY, os.ModePerm)
+		if err != nil {
+			return err
+		}
+		defer fh.Close()
+
+		// Cap last step to available blocks.
+		if last < i+step {
+			step = last - i
+		}
+
+		w := era.NewBuilder(fh)
+		for j := uint64(0); j < step; j++ {
+			nr := i + j
+			block := bc.GetBlockByNumber(nr)
+			if block == nil {
+				return fmt.Errorf("export failed on #%d: not found", nr)
+			}
+			receipts := bc.GetReceiptsByHash(block.Hash())
+			if receipts == nil {
+				return fmt.Errorf("export failed on #%d: receipts not found", nr)
+			}
+			if err := w.Add(block, receipts); err != nil {
```

Comment:
what about withdrawals?

Reply:
This is specifically for pre-merge data, not sure if it makes sense to also support withdrawals? I guess it doesn't really hurt.